api/src/main/java/uk/gov/dvsa/mot/motr/service/MotrVehicleHistoryProvider.java

```diff
@@ -0,0 +1,140 @@
+            }
+        } else {
+            throw new InvalidResourceException("No MOT Test or DVLA vehicle found for registration " + registration);
+        }
+    }
+
+    @NotNull public VehicleWithLatestTest searchForCommercialVehicleByRegistration(@NotNull String registration) throws Exception {
+        Optional<VehicleWithLatestTest> dvlaVehicleOptional = motrReadService.getLatestMotTestForDvlaVehicleByRegistration(registration);
+
+        if (shouldGetHgvPsvHistory(dvlaVehicleOptional.isPresent(), registration)) {
+
+            Vehicle hgvPsvVehicle = hgvVehicleProvider.getVehicle(registration);
+
+            if (hgvPsvVehicle != null) {
+                return new HgvPsvVehicleWithLatestTest(hgvPsvVehicle,
+                        dvlaVehicleOptional.map(VehicleWithLatestTest::getDvlaVehicleId).orElse(null));
+            } else {
+                logger.debug("No HGV/PSV vehicle retrieved");
+                throw new InvalidResourceException(String.format("No HGV/PSV vehicle found for registration %s", registration));
+            }
+
+        } else {
+            throw new InvalidResourceException(String.format("No DVLA vehicle found for registration %s", registration));
+        }
+    }
+
+    @NotNull public VehicleWithLatestTest getDvlaVehicleWithTestByDvlaVehicleId(Integer dvlaVehicleId)
+    throws TradeException {
+        Optional<VehicleWithLatestTest> vehicle = motrReadService.getLatestMotTestByDvlaVehicleId(dvlaVehicleId);
+
+        return vehicle.orElseThrow(() ->
+                new InvalidResourceException(String.format("No MOT Test or DVLA vehicle found for DVLA vehicle id %d", dvlaVehicleId))
+        );
+
+    }
+
+    @NotNull public VehicleWithLatestTest getLatestMotTestByMotTestNumberWithSameRegistrationAndVin(Long motTestNumber)
+    throws TradeException {
+        Optional<VehicleWithLatestTest> vehicle = motrReadService.getLatestMotTestByMotTestNumberWithSameRegistrationAndVin(motTestNumber);
+        return vehicle.orElseThrow(() ->
+            new InvalidResourceException(String.format("No MOT Tests found with number: %d", motTestNumber))
+        );
+    }
+
+    private VehicleWithLatestTest getVehicleDetailsFromMostRecentSource(VehicleWithLatestTest motVehicle) throws Exception {
+        if (!hasHgvPsvCompatibileVehicleClass(motVehicle) || motVehicle.getDvlaVehicleId() == null) {
+            return motVehicle;
+        }
+
+        Vehicle hgvVehicle = hgvVehicleProvider.getVehicle(motVehicle.getRegistration());
+        if (hgvVehicle == null) {
+            return motVehicle;
+        }
+        return selectMostRecentVehicleHistory(motVehicle, new HgvPsvVehicleWithLatestTest(hgvVehicle, motVehicle.getDvlaVehicleId()));
+    }
+
+    private boolean hasHgvPsvCompatibileVehicleClass(VehicleWithLatestTest motVehicle) {
+        String vehicleClass =  motVehicle.getMotVehicleClass();
```

Comment:
Double spacebar after `=` 😈

Reply:
how could checkstyle miss this ...
fixed ;)